apps/arweave/src/ar_mining_worker.erl

```diff
@@ -13,29 +13,43 @@
 -include_lib("eunit/include/eunit.hrl").
 
 -record(state, {
-	name						= not_set,
-	partition_number			= not_set,
-	diff_pair					= not_set,
-	packing_difficulty			= 0,
-	task_queue					= gb_sets:new(),
-	active_sessions				= sets:new(),
+	name = not_set,
+	partition_number = not_set,
+	diff_pair = not_set,
+	packing_difficulty = 0,
+	task_queue = gb_sets:new(),
 	%% The sub_chunk_cache stores either the first or second sub-chunk for a given nonce. This
 	%% is because we process both the first and second recall ranges in parallel and don't know
 	%% which data will be available first. For spora_2_6 packing (aka difficulty 0), sub-chunks
-	%% and chunks are the same size (256KiB), for composite packing each sub-chunk is 8KiB.
-	sub_chunk_cache 			= #{},
-	sub_chunk_cache_size		= #{},
-	sub_chunk_cache_limit		= 0,
-	vdf_queue_limit				= 0,
-	latest_vdf_step_number		= 0,
-	is_pool_client				= false,
-	h1_hashes					= #{},
-	h2_hashes					= #{}
+	%% and chunks are the same size (256KiB), for replica packing each sub-chunk is 8KiB.
+	chunk_cache = undefined,
+	vdf_queue_limit = 0,
+	latest_vdf_step_number = 0,
+	is_pool_client = false,
+	h1_hashes = #{},
+	h2_hashes = #{}
 }).
 
 -define(TASK_CHECK_FREQUENCY_MS, 200).
 -define(STATUS_CHECK_FREQUENCY_MS, 5000).
 
+%%%===================================================================
+%%% Messages
+%%%===================================================================
+
+-define(MSG_RESET_DIFFICULTY(DiffPair), {reset_difficulty, DiffPair}).
```

Comment:
I think the reason we went with `reset` rather than `reset_difficulty` is that the active session list is also cleared vs. `set_difficulty` that only sets the dificulty.

i.e. `reset` is called when there's a mining session reset and `set_difficulty` is called when it's only the difficult that's being updated without reseting the mining session

Reply:
Got it.